Userland/ls.cpp

```diff
@@ -114,6 +116,8 @@ int print_name(struct stat& st, const char* name, const char* path_for_link_reso
 
 bool print_filesystem_object(const char* path, const char* name) {
     struct stat st;
+    struct passwd *pw;
```

Comment:
This is C++, not old C, so we should just declare the variables below, when they are first used, instead of keeping them around uninitialized for a while.